app/models/user.rb

```diff
@@ -1,10 +1,115 @@
+require "httpclient"
+require "json"
+
+# days counted from 00:00 of the first of every month
+DAYS_PASSED_PER_MONTH = {
+  "1" => { normal: 0, leap: 0 },
+  "2" => { normal: 31, leap: 31 },
+  "3" => { normal: 59, leap: 60 },
+  "4" => { normal: 90, leap: 91 },
+  "5" => { normal: 120, leap: 121 },
+  "6" => { normal: 151, leap: 152 },
+  "7" => { normal: 181, leap: 182 },
+  "8" => { normal: 212, leap: 213 },
+  "9" => { normal: 243, leap: 244 },
+  "10" => { normal: 273, leap: 274 },
+  "11" => { normal: 304, leap: 305 },
+  "12" => { normal: 334, leap: 335 }
+}
+DAYS_PASSED_PER_MONTH.freeze # ???
+
+# days passed since J2000 epoch as of 00:00 01/01/16
+DAYS_SINCE_J2000 = 5842.5
+
 class User < ActiveRecord::Base
 
-  has_many :stars
+  has_many :histories
+  has_many :stars, through: :histories
+
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
   validates :username, presence: true, uniqueness: true
 
+  def set_location(ip)
```

Comment:
Unused method argument - `ip`. If it's necessary, use `_` or `_ip` as an argument name to indicate that it won't be used. You can also write as `set_location(*)` if you want the method to accept any arguments but don't care about them.